packages/core/test/actor.test.ts

```diff
@@ -1177,4 +1177,49 @@ describe('actors', () => {
 
     expect(service.getSnapshot().value).toBe('done');
   });
+
+  it('should not restart a completed observable', (done) => {
+    let subscriptionCount = 0;
+    const machine = createMachine({
+      invoke: {
+        id: 'observable',
+        src: fromObservable(() => {
+          const obs = of(42);
+          const sub = obs.subscribe.bind(obs);
+
+          obs.subscribe = (...args) => {
+            subscriptionCount++;
+            return sub(...(args as any));
+          };
+
+          return obs;
+        })
+      }
+    });
+
+    const actor = interpret(machine).start();
+
+    const obs = actor
+      .getSnapshot()
+      .children['observable'].getPersistedState?.();
+
+    expect(obs).toEqual(
+      expect.objectContaining({
+        data: 42,
+        status: 'done'
+      })
+    );
+
+    const persistedState = actor.getPersistedState();
+
+    interpret(machine, {
+      state: persistedState
+    }).start();
+
+    setTimeout(() => {
+      // Will be 2 if the observable is resubscribed
+      expect(subscriptionCount).toBe(1);
+      done();
+    });
```

Comment:
everything here is sync, there is no need to complicate this with async jobs:
```suggestion
    // Will be 2 if the observable is resubscribed
    expect(subscriptionCount).toBe(1);
```